node/spr.go

```diff
@@ -31,21 +31,36 @@ func (d *Pegnetd) GradeS(ctx context.Context, block *factom.EBlock) (graderStake
 	if block.Height >= config.V202EnhanceActivation {
 		ver = 7
 	}
+	if block.Height >= config.PIP18DelegateStakingActivation {
+		ver = 8
+	}
 
 	g, err := graderStake.NewGrader(ver, int32(block.Height))
-	if err != nil {
+	gv2, err2 := graderStake.NewGraderV4(ver, int32(block.Height))
+	if err != nil && ver < 8 {
 		return nil, err
 	}
+	if err2 != nil && ver >= 8 {
+		return nil, err2
+	}
 	for _, entry := range block.Entries {
 		extids := make([][]byte, len(entry.ExtIDs))
 		for i := range entry.ExtIDs {
 			extids[i] = entry.ExtIDs[i]
 		}
 		// allow only top 100 stake holders submit prices
 		stakerRCD := extids[1]
-		if d.Pegnet.IsIncludedTopPEGAddress(stakerRCD) {
-			// ignore bad opr errors
-			err = g.AddSPR(entry.Hash[:], extids, entry.Content)
+		if block.Height < config.PIP18DelegateStakingActivation {
+			if d.Pegnet.IsIncludedTopPEGAddress(stakerRCD) {
+				// ignore bad opr errors
+				err = g.AddSPR(entry.Hash[:], extids, entry.Content)
+				if err != nil {
+					// This is a noisy debug print
+					//logrus.WithError(err).WithFields(logrus.Fields{"hash": entry.Hash.String()}).Debug("failed to add spr")
+				}
+			}
+		} else if pegBalance, _ := d.Pegnet.GetPEGAddress(stakerRCD); pegBalance > 0 {
```

Comment:
At this line, pegBalance is just main staker's balance. (One person's balance).
Instead of this, sum of delegators PEG balance should be used so that delegation pool receive the staking rewards for its delegators too.
1. Need to read all delegators signatures data from `entry.Content`.
2. Verify Signature using `primitives.VerifySignature()`.
3. Read every PEG balance of delegators based on verification.
4. Send sum of balances to `AddSPRV4()` function